frontend/src/store.js

```diff
@@ -10,22 +10,106 @@ const MUTATIONS = Object.freeze({
 
 export default new Vuex.Store({
   state: {
-    students: []
+    students: [],
   },
   mutations: {
     [MUTATIONS.SET_STUDENTS](state, students) {
       state.students = students
-    }
+    },
+
   },
   actions: {
     async getAllStudents({ commit }) {
       const response = await fetch("/api/students")
       const data = await response.json()
       commit(MUTATIONS.SET_STUDENTS, data)
-    }
+    },
+
+    // Update student status
+    async updateStudentStatus({ commit, dispatch }, { studentID, previousStatusString, nextStatusString, updatedBy }) {
+      var success = true;
+
+      // PATCH student
+      const studentRequestOptions = {
+        method: "PATCH",
+        headers: {
+          'Accept': 'application/json',
+          'Content-Type': 'application/json'
+        },
+        body: JSON.stringify({StudentID: studentID, StatusString: nextStatusString})
+      }
+
+      fetch("/api/students/" + studentID, studentRequestOptions)
+        .then(
+          function(response) {
+            // If PATCH fails, return
+            if(response.status !== 200) {
+              success = false;
+              return;
+            }
+          }
+        )
+
+      // POST statusUpdate
+      const statusUpdateRequestOptions = {
+        method: "POST",
+        headers: {
+          'Accept': 'application/json',
+          'Content-Type': 'application/json'
+        },
+        body: JSON.stringify({
+          StudentID: studentID,
+          PreviousStatusString: previousStatusString,
+          NextStatusString: nextStatusString,
+          UpdatedBy: updatedBy
+        })
+      }
+      fetch("/api/statusUpdates", statusUpdateRequestOptions)
+        .then(
+          function(response) {
+            // If PATCH fails, return
+            if(response.status !== 200) {
+              success = false;
+              return;
+            }
+          }
+        )
+
+      // Check if both PATCH student and POST statusUpdate succeed
+      if (success) { dispatch('getAllStudents') }
+    },
+
+    // Update student English proficiency
+    async updateStudentEnglishProficiency({ commit, dispatch }, { studentID, englishProficiency }) {
+      var success = true;
+
+      // PATCH student
+      const studentRequestOptions = {
+        method: "PATCH",
+        headers: {
+          'Accept': 'application/json',
+          'Content-Type': 'application/json'
+        },
+        body: JSON.stringify({StudentID: studentID, EnglishProficiency: englishProficiency})
+      }
+
+      fetch("/api/students/" + studentID, studentRequestOptions)
+        .then(
+          function(response) {
+            // If PATCH fails, return
+            if(response.status !== 200) {
+              success = false;
+              return;
+            }
+          }
+        )
 
+      // Check if both PATCH student and POST statusUpdate succeed
```

Comment:
Just a nit but this comment is not totally accurate. There is only one operation here.

Reply:
Good catch! Edited the comment ..